src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+  static deleteIndex = Animal.alive.findIndex((animal) => animal.health <= 0);
+  constructor(health, name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  deleteAnimale() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name, hidden = false) {
```

Comment:
The `health` parameter is unnecessary in the `Herbivore` constructor for the same reason as in the `Animal` class. The `hidden` parameter should not have a default value here since it's always initialized to `false` in the `Herbivore` class.